bin.src/generateCommands.py

```diff
@@ -0,0 +1,581 @@
+#!/usr/bin/env python
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import argparse
+import os
+import re
+import shlex
+import sys
+import textwrap
+
+import yaml
+
+from typing import Any, Dict, Iterator, List, Mapping, Optional, Sequence, Tuple
+
+
+def main():
+    parser = argparse.ArgumentParser(description="""
+    Process a yaml file and generate the desired calibrations
+    """)
+
+    parser.add_argument('dataDir', type=str, help="Root of data repository")
+    parser.add_argument('specFile', type=str, help="Name of file specifying the work")
+    parser.add_argument('--bootstrap', action="store_true", help="Bootstrap the calibrations?")
+    parser.add_argument('--blocks', type=str, nargs="+", help="Blocks to execute")
+    parser.add_argument('--calib', type=str, help="Name of output calibration directory", default="CALIB")
+    parser.add_argument('--calibTypes', type=str, nargs="+", help="Types of calibs to process")
+    parser.add_argument("--copyMode", choices=["move", "copy", "link", "skip"], default="link",
```

Comment:
I don't think `link` is always the best. `copy` is probably the best default, but I'm concerned that the best choice may change as a function of calib type.

Reply:
I changed the default option to `copy`. This option has ceased to influence the copy mode of the initial detectorMap. It is now always `copy` no matter what this option is.